aten/src/ATen/native/cpu/ComplexKernel.cpp

```diff
@@ -0,0 +1,32 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorFactories.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+namespace at {
+namespace native {
+namespace {
+
+void complex_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.input_dtype(), "complex_cpu", [&]() {
+    cpu_kernel(iter, [=](scalar_t a, scalar_t b) -> c10::complex<scalar_t> {
+      return c10::complex<scalar_t>(a, b);
+    });
```

Comment:
cc. @VitalyFedyunin do you have an idea of how we can add vectorization here? 
```
[=](Vec256<scalar_t> a, Vec256<scalar_t> a) -> std::pair<Vec256<c10::complex<scalar_t>>, Vec256<c10::complex<scalar_t>>> { ... });
```
this wouldn't work with the current design of `cpu_kernel` right?

Reply:
It should be `[=](Vec256<scalar_t> a, Vec256<scalar_t> a) -> Vec256<c10::complex<scalar_t>> { ... };` instead

Reply:
Sorry I take it back, you will need custom loop which is to complex to use here.

Reply:
Could you use a kernel to interleave the float values, then reinterpret the storage as complex? 

That is you have inputs a0 a1 a2 and b0 b1 b2, the kernel interleaves to a0 b0 a1 b1 a2 b2, and then outside the kernel the storage is reinterpreted as complex.

Reply:
yeah I like that idea however we would still need a custom kernel since we will still have two `Vec256<c10::complex<scalar_t>>` vectors as output for two `Vec256<scalar_t>` vector input.

Reply:
@wjgan7 just to be clear, adding vectorization for `torch.complex` and `torch.polar` is out of the scope of this PR and can be added in a follow up PR.